Assets/MRTK/SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs

```diff
@@ -877,13 +886,18 @@ private void ResolveLayout()
 
         private void OnEnable()
         {
-            // Register for global input events
             if (CoreServices.InputSystem != null)
             {
-                CoreServices.InputSystem.RegisterHandler<IMixedRealityInputHandler>(this);
-                CoreServices.InputSystem.RegisterHandler<IMixedRealityTouchHandler>(this);
-                CoreServices.InputSystem.RegisterHandler<IMixedRealityPointerHandler>(this);
-                CoreServices.InputSystem.RegisterHandler<IMixedRealitySourceStateHandler>(this);
+                // Register for event propagation on trickle down phase in order to handle events before children
+                CoreServices.InputPropagationSystem?.RegisterPropagationHandler<IMixedRealityInputHandler>(this, PropagationPhase.TrickleDown);
+                CoreServices.InputPropagationSystem?.RegisterPropagationHandler<IMixedRealityInputHandler<Vector2>>(this, PropagationPhase.TrickleDown);
+                CoreServices.InputPropagationSystem?.RegisterPropagationHandler<IMixedRealityInputHandler<Vector3>>(this, PropagationPhase.TrickleDown);
+                CoreServices.InputPropagationSystem?.RegisterPropagationHandler<IMixedRealityInputHandler<MixedRealityPose>>(this, PropagationPhase.TrickleDown);
+                CoreServices.InputPropagationSystem?.RegisterPropagationHandler<IMixedRealityTouchHandler>(this, PropagationPhase.TrickleDown);               
+
+                // Register for global input events
+                CoreServices.InputSystem?.RegisterHandler<IMixedRealitySourceStateHandler>(this);
```

Comment:
Curious as to why we still need to listen to global input events. Any idea why we still need to be global input listeners?

Reply:
This is only temporary until scroll has its own interaction collider to capture pointer and source events ( even when all child button colliders are disabled for being out of sight ).